File: 1_Design principles & Patterns/2_Implementing the Factory Method Pattern/Code/FactoryMethodPatternExample/src/Word.java
import java.util.Scanner;

public class Word implements Document {
    private String Title;
    private String Content;

    @Override
    public void create() {
        System.out.println("Opening Word document...");
    }

    @Override
    public void initialize() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter title for Word document: ");
        this.Title = sc.nextLine();
        System.out.print("Enter content for Word document: ");
        this.Content = sc.nextLine();
    }

    @Override
    public void show(){
        System.out.println(Title);
        System.out.println("\n  \n" + Content);
    }

}

File: 1_Design principles & Patterns/2_Implementing the Factory Method Pattern/Code/FactoryMethodPatternExample/FactoryMethodPatternExample/Word.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FactoryMethodPatternExample_
{
    public class Word : IDocument
    {
        private string Title;
        private string Content;

        public void Create()
        {
            Console.WriteLine("Opening Word document...");
        }

        public void Initialize()
        {
            Console.Write("Enter title for Word document: ");
            Title = Console.ReadLine();
            Console.Write("Enter content for Word document: ");
            Content = Console.ReadLine();
        }

        public void Show()
        {
            Console.WriteLine("\nTitle: " + Title);
            Console.WriteLine("Content:\n" + Content);
        }
    }
}
